src/app/[lng]/quests/[slug]/page.tsx

```diff
@@ -1,22 +1,63 @@
-// todo: adjust metadata for quests-page
-// export const metadata: Metadata = {
-//   title: 'Jumper | Quests',
-//   description: 'Dive into the Quests',
-//   alternates: {
-//     canonical: `${process.env.NEXT_PUBLIC_SITE_URL}/quests/`,
-//   },
-// };
-
+import type { Metadata } from 'next';
+import { notFound } from 'next/navigation';
+import { siteName } from 'src/app/lib/metadata';
+import { getSiteUrl } from 'src/const/urls';
+import type { QuestAttributes } from 'src/types/loyaltyPass';
+import { sliceStrToXChar } from 'src/utils/splitStringToXChar';
 import { getQuestBySlug } from '../../../lib/getQuestBySlug';
 import QuestPage from '../../../ui/quests/QuestMissionPage';
-import { notFound } from 'next/navigation';
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { slug: string };
+}): Promise<Metadata> {
+  try {
+    const quest = await getQuestBySlug(params.slug);
+
+    if (!quest || !quest.data) {
+      throw new Error();
+    }
+
+    const questData = quest.data.attributes as QuestAttributes;
+
+    const openGraph: Metadata['openGraph'] = {
+      title: `Jumper Quest | ${sliceStrToXChar(questData.Title, 45)}`,
+      description: `${sliceStrToXChar(questData.Information || 'Quest description', 60)}`,
+      siteName: siteName,
+      url: `${getSiteUrl()}/quests/${params.slug}`,
+      images: [
+        {
+          url: `${quest.url}${questData.Image.data.attributes?.url}`,
+          width: 900,
+          height: 450,
+          alt: 'banner image',
+        },
+      ],
+      type: 'article',
+    };
+
+    return {
+      title: `Jumper Quest | ${sliceStrToXChar(questData.Title, 45)}`,
+      description: questData.Subtitle,
+      alternates: {
+        canonical: `${getSiteUrl()}/quests/${params.slug}`,
+      },
+      twitter: openGraph,
+      openGraph,
+    };
+  } catch (err) {
+    return {
+      title: `Jumper Quest | ${sliceStrToXChar(params.slug.replaceAll('-', ' '), 45)}`,
+      description: `This is the description for the quest "${params.slug.replaceAll('-', ' ')}".`,
+    };
+  }
+}
 
 export default async function Page({ params }: { params: { slug: string } }) {
   const { data, url } = await getQuestBySlug(params.slug);
-
-  if (!data?.data?.[0]) {
+  if (!data) {
```

Comment:
Might not be enough, data can be returned and data.data -> [] but will still not go through

Reply:
While adding the proper types on our fetch function `getQuestBySlug` I have also tweaked the return statement to the following (see screen). If there is no returned data, it should actually be undefined -->

![image](https://github.com/user-attachments/assets/a7e646c9-6493-417f-b88e-fdfa6dfb2436)